[Feature] - Custom deprecation (#6005)
OpenBB-finance/OpenBBTerminal: website/content/platform/development/contributor-guidelines/deprecating_endpoints.md

---
title: Deprecating Endpoints
sidebar_position: 5
description: This guide provides detailed instructions on how to deprecate an endpoint in the OpenBB Platform.
keywords:
- OpenBB community
- OpenBB Platform
- Custom commands
- API
- Python Interface
- Deprecation
- Deprecated
---

import HeadTitle from '@site/src/components/General/HeadTitle.tsx';

<HeadTitle title="Deprecating Endpoints - Contributor Guidelines - Development | OpenBB Platform Docs" />

> Deprecating commands is a necessary part of maintaining the OpenBB Platform. This guide outlines the process for deprecating an endpoint.

Reviewer: ```suggestion
Deprecating commands is essential to maintaining the OpenBB Platform. This guide outlines the process for deprecating an endpoint.
```

## Deprecating an endpoint

> 1. Add new endpoint: the new endpoint that will replace the deprecated one.

Reviewer: ```suggestion
1. Add the new endpoint that will replace the deprecated one.
```

> 2. Add deprecation warning

Reviewer: ```suggestion
2. Add the deprecation warning
```

>     Navigate to the **router** where the endpoint you wish to deprecate lives and add the `deprecated=True` flag and `deprecation=OpenBBDeprecationWarning(…)`, to the `@command` decorator. Refer to the example below:

Reviewer: ```suggestion
    Navigate to the **router** where the endpoint to be deprecated exists. Set the `deprecated` flag to `True` and add `deprecation=OpenBBDeprecationWarning(…)` argument to the decorator. Refer to the example below:
```

    ```python

    ...
    from openbb_core.app.deprecation import OpenBBDeprecationWarning
    ...

        @router.command(
            model="MarketIndices",
            deprecated=True,
            deprecation=OpenBBDeprecationWarning(
                message="This endpoint is deprecated; use `/index/price/historical` instead.",
                since=(4, 1),
                expected_removal=(4, 5),
            ),
        )
        async def market(
            cc: CommandContext,
            provider_choices: ProviderChoices,
            standard_params: StandardParams,
            extra_params: ExtraParams,
        ) -> OBBject[BaseModel]:
            """Historical Market Indices."""
        return await OBBject.from_query(Query(**locals()))

    ```

> 3. Get approval from a OpenBB Platform maintainer: they will help you determine the appropriate version for the deprecation warning, and also know what to do in order to warn other OpenBB teams that might depend on the endpoint you are deprecating.

Reviewer: ```suggestion
3. Get approval from a OpenBB Platform maintainer: We will help you determine the appropriate version for the deprecation warning, and communicate the change(s) to the relevant personnel that might depend on the endpoint you are deprecating.
```
